internal/releasesources/clients.go

```diff
@@ -0,0 +1,146 @@
+package releasesources
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+)
+
+const releaseImagesResourcePath = "upgrades_info/v1/graph"
+
+// OCPVersionSupportLevels maps between major.minor OCP version to its corresponding suppurt level
+type OCPVersionSupportLevels map[string]string
+
+type OCPMajorVersionSet map[string]bool
+
+//go:generate mockgen -source=clients.go -destination=mock_clients.go -package=releasesources
+type OpenShiftReleasesAPIClientInterface interface {
+	getEndpoint(channel, openshiftVersion, cpuArchitecture string) string
+	parseResponse(response *http.Response) (*releaseGraph, error)
+	GetReleases(channel, openshiftVersion, cpuArchitecture string) (*releaseGraph, error)
+}
+
+//go:generate mockgen -source=clients.go -destination=mock_clients.go -package=releasesources
+type OpenShiftSupportLevelAPIClientInterface interface {
+	getEndpoint(openshiftMajorVersion string) string
```

Comment:
Why we need private methods in interface?

Reply:
I think the confusion comes from the fact I wrote this interface as public instead of private, it is intended only for this package